app/src/test/java/com/codesoom/assignment/application/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,85 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.errors.InvalidAccessTokenException;
+import com.codesoom.assignment.utills.JwtUtill;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("사용자 권한 서비스")
+class AuthenticationServiceTest {
+    private static final String SECRET = "12345678901234567890123456789010";
+
+    private static final String VALID_TOKEN = "eyJhbGciOiJIUzI1NiJ9." +
+            "eyJ1c2VySWQiOjF9.ZZ3CUl0jxeLGvQ1Js5nG2Ty5qGTlqai5ubDMXZOdaDk";
+    private static final String INVALID_TOKEN = "eyJhbGciOiJIUzI1NiJ9." +
+            "eyJ1c2VySWQiOjF9.ZZ3CUl0jxeLGvQ1Js5nG2Ty5qGTlqai5ubDMXZOdaD0";
```

Comment:
INVALID_TOKEN을 쉽게 만드는 방법이 하나 있습니다.
```suggestion
    private static final String INVALID_TOKEN = VALID_TOKEN + "$$$";
```

Reply:
좋은 방법이네요! 
`private static final String INVALID_TOKEN = VALID_TOKEN + "$$$";`
로 작성하면 에러가 나고 
`private static final String INVALID_TOKEN = VALID_TOKEN + "특수문자아닌 문자열";`로 테스트하면 통과됩니다.

- Base64URL : URL 주소로 사용가능하도록 URL에서 특별한 의미를 가지는 특수문자를 제외한 Base64
- Base64 : 8비트 이진 데이터를 ASCII 영역 문자로만 이뤄진 문자열로 바꾸는 인코딩 방식

때문에 그런것인가요? header,payload 는 Base64URL이고 signature부분도 그런가요?

Reply:
`$$$`는 아무렇게나 생각나는대로 적은 거에요. 적절한 문자를 붙여주시면 됩니다. 시그니처 부분도 base64url을 사용합니다.
![image](https://user-images.githubusercontent.com/1855714/148217127-87d33260-2b29-43bc-832b-9cfa78e43645.png)

JWT가 정의된 RFC 7519를 읽어보세요.
https://datatracker.ietf.org/doc/html/rfc7519

이번 기회에 base64에 대해 알아보는 것도 의미있는 공부가 되겠네요. 

아래는 제가 예전에 작성해 둔 글입니다.
https://johngrib.github.io/wiki/base64/